x-pack/plugins/observability_solution/inventory/public/pages/inventory_page/index.tsx

```diff
@@ -4,105 +4,73 @@
  * 2.0; you may not use this file except in compliance with the Elastic License
  * 2.0.
  */
-import { EuiDataGridSorting } from '@elastic/eui';
-import React from 'react';
-import useEffectOnce from 'react-use/lib/useEffectOnce';
-import { EntityColumnIds, EntityType } from '../../../common/entities';
-import { EntitiesGrid } from '../../components/entities_grid';
-import { useInventorySearchBarContext } from '../../context/inventory_search_bar_context_provider';
-import { useInventoryAbortableAsync } from '../../hooks/use_inventory_abortable_async';
+import React, { useCallback, useEffect, useState } from 'react';
+import { decode, encode } from '@kbn/rison';
+import { UngroupedInventoryPage } from '../../components/grouped_inventory/ungrouped';
+import { GroupedInventoryPage } from '../../components/grouped_inventory/grouped';
 import { useInventoryParams } from '../../hooks/use_inventory_params';
 import { useInventoryRouter } from '../../hooks/use_inventory_router';
-import { useKibana } from '../../hooks/use_kibana';
+import { InventoryPageViewContextProvider } from '../../context/inventory_page_view_provider';
+
+const DEFAULT_VIEW = 'type';
+
+export interface InventoryState {
+  grouping?: string;
```

Comment:
Is grouping the **group by field** the user selected?

Reply:
Yes, so if the user selects "type", then it is by `entity.type`, else it is "none" which is the ungrouped view